### ISSUE TITLE
$defineI18nRoute won't match if localRoutes passed by variable

### ISSUE THREAD

**Issue by Buzut** [s00d/nuxt-i18n-micro#32]
**Describe the bug**
`$defineI18nRoute` doesn't seem to accept anything else than string literals.

**To Reproduce**
In `pages/product/[slug].vue`

```js
// This works as expected
$defineI18nRoute({
  localeRoutes: {
    de: '/produkt/:slug()',
    en: '/product/:slug()',
    es: '/producto/:slug()',
  }
});

// This won't work (404)
const locales = {
    de: '/produkt/:slug()',
    en: '/product/:slug()',
    es: '/producto/:slug()',
  };

$defineI18nRoute({
  localeRoutes: locales
});

// This won't work either (404)
const locales = { localeRoutes: {
    de: '/produkt/:slug()',
    en: '/product/:slug()',
    es: '/producto/:slug()',
}};

$defineI18nRoute(locales);
```

**Expected behavior**
Should work exactly the same way.

**Comment by Buzut** [comment]
Not sure what's the issue, is it serialised and then transmitted to H3? I confess I didn't try to pinpoint the exact spot in the codebase where the issue would arise.

Unlike the other matter we're discussing, this one isn't blocking. But it leads to some boilerplate that could be nicely addressed by local helpers. At first I wanted to create a helper to declare my urls, kind of like so:

```js
import mapSlug from mapSlug; // this just returns corresponding localised slug
 
export default function (slug) {
  const routes = {};
  $getLocales().forEach((locale) => {
    routes[locale.code] = `/${mapSlug(slug, locale.code)}/:slug()`;
  });

  return { localeRoutes: { routes }};
}
```

Maybe some help from the good old `eval()` could make it 🙈

**Comment by s00d** [comment]
The issue arises due to server-side rendering (SSR). During the build process, the project requires already generated links, and my module searches for defineI18nRoute using regex and simply returns a JSON structure. There are many other approaches that could partially help, such as executing JavaScript during the build. However, this would introduce another major problem: during the build, there is no Nuxt context, which means methods like getLocales in your example physically cannot work.

To implement a solution for this, we would either need to render a Nuxt page at build time to retrieve the necessary data or integrate defineI18nRoute into Nuxt’s core. The second option is something I’d love to explore myself, but I’m not sure yet how to make it work.

**Comment by Buzut** [comment]
We can maybe strike a middle ground by just executing JS without the Nuxt context as it might be overkill. If needed, one can manually load the Nuxt config file if needed. IMHO, it's totally acceptable.

**Comment by s00d** [comment]
I looked into how others do it, for example, the Nuxt developers, and realized they don’t really handle it.  
[Link](https://github.com/nuxt/nuxt/blob/560f176c097db497c91aeed3071fe8642e089e38/packages/nuxt/src/pages/utils.ts#L185)  
They just parse the code, and if it's not a string, they throw an error. I did a few experiments and wrote some semi-functional code, but I don’t like it. I'll keep studying it further.

**Comment by Buzut** [comment]
Just the fact you're taking the time to consider it is already super cool! No worries if can't be done in a clean way.
In that case it's not an issue, but by design.

**Comment by s00d** [comment]
Unfortunately, I still haven't found a working solution.

**Comment by Buzut** [comment]
Yeah I can live with that. I’d rather have an efficient tool that does slightly less than a bloated one that tries to do everything. 
Thank you anyway 🙌
